docs/architecture/architecture.md

```diff
@@ -609,6 +614,117 @@ rootfs by calling `mount(8)` within the counter.  If the devicemapper block devi
 is used, `/` will be mounted on `/dev/vda`.  Users can disable direct mounting
 of the underlying block device through the runtime configuration.
 
+## Kubernetes support
+[Kubernetes\*](https://github.com/kubernetes/kubernetes/) is the dominant open source
+container orchestration engine. It runs a control plane where a scheduler
+(typically running on a dedicated master node) calls into a compute node container
+agent called `kubelet`. On each compute node a `kubelet` instance is responsible
+for managing the node pods lifecycle and eventually relies on a container runtime for
+handling the container execution parts. The `kubelet` architecture decouples the
+lifecycle management part from the container execution one through the dedicated `gRPC`
```

Comment:
consider hot-linking `gRPC` to maybe the gRPC definition or docs (either generic google ones or specific k8s/CRI-O ones if such a thing exists)